matching-service/src/routes/routes.ts

```diff
@@ -41,49 +42,54 @@ router.post("/start_matching", async (req : Request, res : Response) => {
     }
     
     try {
-        // Send the new user to the Kafka topic
-        await sendQueueingMessage(user.userToken);
-        userStore.addUser(user.userToken, user);
-
-        // Listen to matching outcome
-        // await listenForMatchResult(user.userToken, (result) => {
-        //     if (result === 'matched') {
-        //         return res.status(200).send({message: "Match found"});
 
-        //     } else if (result === 'timeout') {
-        //         userStore.removeUser(user.userToken);
-        //         return res.status(408).send({message: "No match found due to timeout"});
-        //     }
-        // });
-        return res.status(200).send({message: "Started Queueing"});
+        // Check if the user is already matching or matched
+        if (userStore.hasUser(user.id)) {
+            return res.status(409).send({message: "This user already matching."});
+        }
+        // Send the new user to the Kafka topic
+        userStore.addUser(user.id, user);
+        await sendQueueingMessage(user.id);
+        
+        return res.status(200).send({message: "Started Queueing User: " + user.email});
 
         // By default, no match is found
-        userStore.removeUser(user.userToken);
-        return res.status(204).send({message: "No match found"});
+        // userStore.removeUser(user.userToken);
+        // return res.status(204).send({message: "No match found"});
```

Comment:
if u think you dont need this, just delete it ba, then in `sendStartMatchingRequest`, remove the check for status `204`